.github/workflows/ci.yaml

```diff
@@ -554,7 +552,7 @@ jobs:
   benchmarks:
     runs-on: ubuntu-latest
     needs: determine_changes
-    if: ${{ needs.determine_changes.outputs.code == 'true' || github.ref == 'refs/heads/main' }}
+    if: ${{ github.repository == 'astral-sh/ruff' && (needs.determine_changes.outputs.code == 'true' || github.ref == 'refs/heads/main') }}
```

Comment:
This should be skipped for forks because our codspeed token is only valid for `astral-sh/ruff`.

This is quite a long line. I think there are ways of doing multiline if-conditions in GitHub Actions, but they seem to be pretty poorly documented, and I've got it wrong frequently in the past. I'd prefer to just do a very long single-line if-condition, as I currently have.